Flow.Launcher.Plugin/Result.cs

```diff
@@ -62,7 +62,13 @@ public string IcoPath
             {
                 if (!string.IsNullOrEmpty(PluginDirectory) && !Path.IsPathRooted(value))
                 {
-                    _icoPath = Path.Combine(value, IcoPath);
+                    string absPath = Path.Combine(value, IcoPath);
+                    // Only convert relative paths if its a valid path
+                    if (File.Exists(absPath))
```

Comment:
Is there another way to check this path without using File.Exists? Doesn't seem necessary to use it here.

Reply:
This way allows us to improve how Flow auto appends the plugins path to icon paths that are relative and still accept URI such as http, https, ftp, etc. instead of checking every URI under the sun to see if it's valid let's just detect if that file even exists and leave it alone if it doesn't.

Reply:
I mean this is an unnecessary call because we already checking that the path exists or not at https://github.com/Flow-Launcher/Flow.Launcher/blob/e5948a70c41362bde75d82c91c9a422b352bbb35/Flow.Launcher.Infrastructure/Image/ImageLoader.cs#L183-L202

(GetThumbnailResult is called by LoadInternalAsync)

We just need to determine that it is not a URI.

Reply:
We *can* detect URI and still keep original function that adds the plugins root directory to relative paths. This way just seemed to be the simplest way to accomplish both.

Should I refactor this to detect URI's and relative paths?

Reply:
We're not really concerned with if the path exists or not. What we're really concerned with is the path really a relative path or not.

The built-in method for detecting relative paths isnt very smart.

Reply:
@Garulf @taooceros I reworked IcoPath, PluginDirectory and ImageLoader https://github.com/Flow-Launcher/Flow.Launcher/pull/1351/commits/83ec8099a78cf64c446b53cefa52ce6e611c7dbf, let me know if ok.